src/main/scala/lc/background/taskThread.scala

```diff
@@ -22,13 +22,21 @@ class BackgroundTask(throttle: Int, timeLimit: Int) {
         if (imageNum.next())
           throttleIn = (throttleIn - imageNum.getInt("total"))
         while (0 < throttleIn) {
-          Captcha.generateChallenge(Parameters("medium", "image/png", "text", Option(Size(0, 0))))
+          Captcha.generateChallenge(getRandomParam())
           throttleIn -= 1
         }
       } catch { case exception: Exception => println(exception) }
     }
   }
 
+  private def getRandomParam(): Parameters = {
+    val level = Config.allowedLevels.toList(Config.getNextRandomInt(Config.allowedLevels.size))
+    val media = Config.allowedMedia.toList(Config.getNextRandomInt(Config.allowedMedia.size))
+    val inputType = Config.allowedInputType.toList(Config.getNextRandomInt(Config.allowedInputType.size))
```

Comment:
define a common method to pick a random element from a list / set:

```
def pickRandom[T](list: List[T]) : T = {
  list(Config.getNextRandomInt(l.size))
}
```

and use it.